src/engine/strat_engine/mdv.rs

```diff
@@ -37,6 +37,51 @@ const FILESYSTEM_DIR: &'static str = "filesystems";
 pub struct MetadataVol {
     dev: LinearDev,
     mount_pt: PathBuf,
+    is_mounted: bool,
+}
+
+/// A helper struct that borrows the MetadataVol and ensures that the MDV is
+/// mounted as long as it is borrowed, and then unmounted.
+#[derive(Debug)]
+struct MountedMDV<'a> {
+    mdv: &'a mut MetadataVol,
+}
+
+impl<'a> MountedMDV<'a> {
+    /// Borrow the MDV and ensure it's mounted.
+    fn mount(mdv: &mut MetadataVol) -> EngineResult<MountedMDV> {
+        if !mdv.is_mounted {
```

Comment:
We think that it might be possible for the volume to be mounted when we think it is not, but we do not allow for the possibility that the volume is unmounted when we think it is. Why is that?

My preference would be to omit ```is_mounted``` and just mount the thing here.

Reply:
Ok.